pkg/controller/cluster/cluster_reconcile_test.go

```diff
@@ -85,6 +85,32 @@ func TestReconcileCluster_Reconcile(t *testing.T) {
 						GitRepoTemplate: &synv1alpha1.GitRepoTemplate{},
 					},
 				},
+				&corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "somesecret",
+						Annotations: map[string]string{
+							corev1.ServiceAccountNameKey: tt.fields.objName,
+						},
+						CreationTimestamp: metav1.Time{Time: time.Now().Add(15 * time.Second)},
+					},
+					Type: corev1.SecretTypeServiceAccountToken,
+					Data: map[string][]byte{
+						"token": []byte("mysecret"),
+					},
+				},
+				&corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "somesecret1",
+						Annotations: map[string]string{
+							corev1.ServiceAccountNameKey: tt.fields.objName,
+						},
+						CreationTimestamp: metav1.Time{Time: time.Now()},
+					},
+					Type: corev1.SecretTypeServiceAccountToken,
+					Data: map[string][]byte{
+						"token": []byte("mysecret1"),
+					},
+				},
```

Comment:
Let's add a test case with a secret and missing `token`.

Reply:
Added a case where the newest secret doesn't have a token and falls back to the previous one.